pkg/topology/kademlia/kademlia.go

```diff
@@ -844,6 +844,11 @@ func binSaturated(oversaturationAmount int, staticNode staticPeerFunc, k *Kad) b
 // recalcDepth calculates, assigns the new depth, and returns if depth has changed
 func (k *Kad) recalcDepth() {
 
+	if k.storageRadius != swarm.MaxPO {
```

Comment:
depth should be minimum of storage radius and connection depth, this early return can set depth to higher than connection depth

Reply:
the reason this change is required, because storage radius starts at reserve radius, if the connection depth is below the radius, then syncing happens at connection depth, and not at storage radius.